modopt/math/matrix.py

```diff
@@ -315,30 +347,34 @@ def get_spec_rad(self, tolerance=1e-6, max_iter=20, extra_factor=1.0):
             (default is ``1.0``)
 
         """
-
         # Set (or reset) values of x.
         x_old = self._set_initial_x()
 
         # Iterate until the L2 norm of x converges.
-        for i in range(max_iter):
+        for i_elem in range(max_iter):
 
             x_old_norm = np.linalg.norm(x_old)
 
             x_new = self._operator(x_old) / x_old_norm
 
-            x_new_norm = np.linalg.norm(x_new)
+            xp = get_array_module(x_new)
```

Comment:
I am unable to do a suggestion here, this must be:
```python
            xp = get_array_module(x_new)

            x_old_norm = xp.linalg.norm(x_old)

            x_new = self._operator(x_old) / x_old_norm

```

Reply:
I guess you mean

```python
xp = get_array_module(x_old)

x_old_norm = xp.linalg.norm(x_old)

x_new = self._operator(x_old) / x_old_norm

x_new_norm = xp.linalg.norm(x_new)
```

as `x_new` is not defined prior to the call to `get_array_module`.

Reply:
Oops, yes you are right!